### ISSUE TITLE
[Bug]: building for openSUSE fails with "unsafe.Slice requires go1.17 or later (-lang was set to go1.16; check go.mod)"

### ISSUE THREAD

**Issue by johanneskastl** [norwoodj/helm-docs#202]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I am packaging this very helpful utility for openSUSE. Currently it fails due to the following error:

```
[    4s] + go build -mod=vendor -buildmode=pie -o bin/helm-docs ./cmd/helm-docs
[    5s] # golang.org/x/sys/unix
[    5s] vendor/golang.org/x/sys/unix/syscall.go:83:16: unsafe.Slice requires go1.17 or later (-lang was set to go1.16; check go.mod)
[    5s] vendor/golang.org/x/sys/unix/syscall_linux.go:1018:20: unsafe.Slice requires go1.17 or later (-lang was set to go1.16; check go.mod)
[    5s] vendor/golang.org/x/sys/unix/syscall_linux.go:2297:9: unsafe.Slice requires go1.17 or later (-lang was set to go1.16; check go.mod)
[    5s] vendor/golang.org/x/sys/unix/syscall_unix.go:118:7: unsafe.Slice requires go1.17 or later (-lang was set to go1.16; check go.mod)
[    5s] vendor/golang.org/x/sys/unix/sysvshm_unix.go:33:7: unsafe.Slice requires go1.17 or later (-lang was set to go1.16; check go.mod)
[    5s] error: Bad exit status from /var/tmp/rpm-tmp.KYRhIP (%build)
```

The go.mod file still states go1.14 as required, I guess this should be increased to go1.17?

(Just changing that one line in the file during build solves this error, but then leads to error due to the vendoring necessary for building without internet connnectivity)

### Expected Behavior

Successful build.

### Reference Chart

None, no runtime problem.

### Environment

Building on openSUSE 64bit

**Comment by Nepo26** [comment]
Hi @johanneskastl,

I was able to reproduce the bug, and it seems like it was a bug in viper on older go versions, like shown in this [issue](https://github.com/spf13/viper/issues/1161). 

So I will look further into it and see if 'go' really needs to be updated to 1.17. 

Furthermore, thank you for the interest in the project! :smile:

**Comment by johanneskastl** [comment]
Any news on this?

**Comment by johanneskastl** [comment]
Fixed with 1.13.0 and go1.22
